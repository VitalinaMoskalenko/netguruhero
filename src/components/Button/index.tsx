import React, { CSSProperties } from "react";
import styled from "styled-components";
import { Body } from "../Paragraphs";

const Container = styled.button`
  height: 40px;
  width: 100%;
  background-color: ${({ theme }) => theme.colors.green};
  border: none;
  border-radius: 8px;
  cursor: pointer;

  :hover {
    opacity: 0.8;
  }
`;

const Text = styled(Body)`
  font-weight: bold;
  color: ${({ theme }) => theme.colors.white};
`;

type PropsType = {
  style?: CSSProperties;
  text: string;
  onClick: () => void;
};

const Button = ({ text, onClick, style }: PropsType) => {
  return (
    <Container style={style} onClick={onClick}>
      <Text>{text}</Text>
    </Container>
  );
};

export default Button;
