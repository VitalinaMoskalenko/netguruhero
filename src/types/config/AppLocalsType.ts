export enum AppLocalsType {
  en = "en",
}
