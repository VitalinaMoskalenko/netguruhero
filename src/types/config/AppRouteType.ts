export enum AppRouteType {
  home = "/",
  addHeroForm = "/addHeroForm",
  heroDetails = "/heroDetails",
  heroDetailsWithId = "/heroDetails/:id",
}
