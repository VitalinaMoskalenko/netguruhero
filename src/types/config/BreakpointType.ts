export enum BreakpointType {
  xxs = 375,
  xs = 768,
  xxl = 1440,
}
