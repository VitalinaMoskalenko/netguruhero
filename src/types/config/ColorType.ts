export enum ColorType {
  white = "white",
  black = "black",
  green = "#63cd93",
  blue = "#6199f3",
  gray = "#f5f6fb",
  black05 = "rgba(0,0,0, 0.5)",
}
