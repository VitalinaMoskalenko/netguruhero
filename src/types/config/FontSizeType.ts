export enum FontSizeType {
  smallBody = 14,
  body = 18,
  h5 = 16,
  h4 = 18,
  h3 = 24,
  h2 = 32,
  h1 = 48,
}
