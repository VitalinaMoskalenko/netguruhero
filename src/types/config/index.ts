export * from "./AppRouteType";
export * from "./FontSizeType";
export * from "./ColorType";
export * from "./BreakpointType";
export * from "./AppLocalsType";
