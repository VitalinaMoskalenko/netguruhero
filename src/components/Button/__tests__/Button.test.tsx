import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import Button from "../";
import { theme } from "../../../config";

it("renders button correctly", () => {
  const buttonComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Button text="Button" onClick={() => {}} />
      </ThemeProvider>
    )
    .toJSON();

  expect(buttonComponent).toMatchSnapshot();
});
