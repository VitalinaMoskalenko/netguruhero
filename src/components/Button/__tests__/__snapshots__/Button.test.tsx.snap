// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<button
  className="sc-dlnjwi hzMCSB"
  onClick={[Function]}
>
  <p
    className="sc-bdnxRM sc-hKFxyN jupkGm gjMbqS"
  >
    Button
  </p>
</button>
`;
